File: src/main/java/com/adinsight/agent/model/AssistantMessage.java
package com.adinsight.agent.model;

import java.util.List;

public record AssistantMessage(String content, List<ToolCall> toolCalls) {
    public AssistantMessage {
        content = content == null ? "" : content;
        toolCalls = toolCalls == null ? List.of() : List.copyOf(toolCalls);
    }

    public boolean needsFollowUp() {
        return !toolCalls.isEmpty();
    }
}

File: src/model/assistant-message.ts
import type { ToolCall } from "./tool-call.js";

export type AssistantMessage = Readonly<{
  content: string;
  toolCalls: readonly ToolCall[];
}>;

export function assistantMessage(input: {
  content?: string | null;
  toolCalls?: readonly ToolCall[] | null;
}): AssistantMessage {
  return {
    content: input.content ?? "",
    toolCalls: [...(input.toolCalls ?? [])]
  };
}

export function needsFollowUp(message: AssistantMessage): boolean {
  return message.toolCalls.length > 0;
}
